src/pyrobot/core.py

```diff
@@ -620,14 +620,29 @@ def set_ee_pose(self, position, orientation, plan=True,
         :return: Returns True if command succeeded, False otherwise
         :rtype: bool
         """
-        joint_positions = self.compute_ik(position, orientation,
-                                          numerical=numerical)
-        result = False
-        if joint_positions is None:
-            rospy.logerr('No IK solution found; check if target_pose is valid')
+        if plan and self.configs.ARM.CLASS != 'LoCoBotArm':
+            if not self.use_moveit:
+                raise ValueError('Using plan=True when moveit is not'
+                                 ' initialized, did you pass '
+                                 'in use_moveit=True?')
+            pose = Pose()
+            position = position.flatten()
+            pose.position.x, pose.position.y, pose.position.z = position[0], position[1], position[2]
+            pose.orientation.x, pose.orientation.y, pose.orientation.z, pose.orientation.w = \
+                self._ori_2_quat(orientation)
+            self.moveit_group.set_pose_target(pose)
+            plan = self.moveit_group.plan()
+            result = self.moveit_group.execute(plan, wait=True)
+            self.moveit_group.clear_pose_targets()
         else:
-            result = self.set_joint_positions(joint_positions,
```

Comment:
Doesn't this already handle obstacles?

Reply:
It only ensures that the trajectory from current joint angle to the goal joint angle doesn't go through obstacle. 
However, the goal joint angle is coming from IK which is not aware of moveit obstacles. So it might find a joint solution which goes through obstacle even though there might be collision free solution available.

Reply:
Hey Dhiraj, `move_group_interface` doesn't have `set_pose_target`, `plan` and `clear_pose_targets` implemented at this point. I modified it to fit the current codebase and tested on sawyer in this PR: https://github.com/facebookresearch/pyrobot/pull/51